docs: add Edge section to security doc (#9562)
Unleash/unleash: website/docs/feature-flag-tutorials/use-cases/security-compliance.md

With Unleash, you can architect your feature flagging solution in a way that all user data stays within your applications. When using server-side SDKs, user data remains within your application and is never shared with the Unleash server regardless of your setup. For client-side SDKs, you can retain all user data within your applications by either [self-hosting Unleash](/understanding-unleash/proxy-hosting#you-host-everything), or [self-hosting Unleash Edge](/understanding-unleash/proxy-hosting#unleash-hosts-the-api-you-host-edge). Read more on the [Unleash architecture here](/understanding-unleash/unleash-overview).

### Use Unleash Edge for enterprise-grade data privacy

To take data privacy, scalability, high performance, and resiliency to the next level for large enterprises, we built [Unleash Edge](/reference/unleash-edge), a lightweight proxy layer between your Unleash API and SDKs. You can deploy Unleash Edge within your own infrastructure so PII and sensitive context data never leave your network boundary.

> Unleash Edge operates as a local evaluation point for feature flags, meaning that the evaluation of flags and user context data happens within your infrastructure. It takes a copy of your feature flag data locally and by default, it uses an in-memory cache to store the features it fetches from your Unleash instance. Another benefit to Edge functioning as a local evaluation point is preventing performance issues in one part of your system from affecting others and maintaining service level agreements required by compliance frameworks.

Reviewer: What is the 'SLAs required by compliance frameworks' referring to? It's not something I've come across before

> Unleash Edge operates as a local evaluation point for feature flags, meaning that the evaluation of flags and user context data happens within your infrastructure. It takes a copy of your feature flag data locally and by default, it uses an in-memory cache to store the features it fetches from your Unleash instance. Another benefit to Edge functioning as a local evaluation point is preventing performance issues in one part of your system from affecting others and maintaining service level agreements required by compliance frameworks.

Reviewer: ```suggestion
Unleash Edge operates as a local evaluation point for feature flags, meaning that the evaluation of flags and user context data happens within your infrastructure. It keeps a local copy of your feature flag configuration in sync with your Unleash instance. Another benefit to Edge functioning as a local evaluation point is preventing performance issues in one part of your system from affecting others and maintaining service level agreements required by compliance frameworks.
```

> Edge allows you to decide what you expose to the world. You only have to expose one project (or a predefined set of projects) and environments through Edge. Nothing else will be exposed. In our cloud, if this is combined with an [IP Allow List](#set-up-ip-allow-lists-for-enhanced-security) for the Unleash instance and Hosted Edge, this means you are reducing the attack surface significantly.

Reviewer: ```suggestion
Edge allows you to decide what you expose to the world. You only have to expose one project (or a predefined set of projects) and environments through Edge. Nothing else will be exposed. In our cloud-hosted offering, you can also define an [IP allow list](#set-up-ip-allow-lists-for-enhanced-security) for the Unleash instance and Hosted Edge to reduce the attack surface.
```

> With Unleash Enterprise Edge, you can keep your core Unleash service hidden from the internet while evaluating feature flags at the edge for better performance and scalability. Each Edge instance can be scoped to specific projects or environments, enforcing the least privileged access and tight network segmentation. Unleash Enterprise Edge cannot access the Unleash Admin API, minimizing the impact of compromised clients or credentials.

Reviewer: ```suggestion
With Unleash Enterprise Edge, you can keep your core Unleash service hidden from the internet while evaluating feature flags at the edge for better performance and scalability. Each Edge instance can be scoped to specific projects or environments, enforcing the least-privileged access and tight network segmentation. Unleash Enterprise Edge cannot access the Unleash Admin API, minimizing the impact of compromised clients or credentials.
```
